calicoctl/calico_ctl/pool.py

```diff
@@ -122,19 +122,58 @@ def ip_pool_add(cidrs, version, ipip, masquerade):
         print "IP in IP not supported for IPv6 pools"
         sys.exit(1)
 
-    # TODO Reject any cidrs that overlap with existing cidrs in the pool
-    pools = []
-    for cidr in cidrs:
+    current_pools = client.get_ip_pools(version)
+    current_pool_nets = [IPNetwork(pool.cidr) for pool in current_pools]
+
+    new_pools = []
+
+    for new_cidr in cidrs:
+        new_cidr_net = IPNetwork(new_cidr)
+
+        # Check if new pool overlaps with any existing pool
+        overlapping_old_cidr = _check_if_cidr_overlaps_cidrs(new_cidr,
+                                   [net.cidr for net in current_pool_nets])
+        if overlapping_old_cidr:
+            print "Cannot add IP pool %s - pool conflicts with exisitng " \
+                  "pool %s" % (new_cidr, overlapping_old_cidr)
+            sys.exit(1)
+
+        # Check if new pool overlaps with any other new pool
+        overlapping_new_cidr = _check_if_cidr_overlaps_cidrs(new_cidr,
+                                   [pool.cidr for pool in new_pools])
+        if overlapping_new_cidr:
+            print "Cannot add IP pool %s - pool conflicts with another " \
+                  "new pool %s" % (new_cidr, overlapping_new_cidr)
+            sys.exit(1)
+
+        # Add cidr as a new IP pool
         try:
-            pools.append(IPPool(cidr, ipip=ipip, masquerade=masquerade))
+            new_pools.append(IPPool(new_cidr, ipip=ipip, masquerade=masquerade))
         except InvalidBlockSizeError:
             print "An IPv%s pool must have a prefix length of %s or lower." \
                   "\nGiven: %s.\nNo pools added." % \
-                  (version, BLOCK_PREFIXLEN[version], cidr)
+                  (version, BLOCK_PREFIXLEN[version], new_cidr)
             sys.exit(1)
-    for pool in pools:
-        client.add_ip_pool(version, pool)
 
+    for new_pool in new_pools:
+        client.add_ip_pool(version, new_pool)
+
+
+def _check_if_cidr_overlaps_cidrs(cidr, cidrs):
+    """
+    Check if a cidr overlaps at all with any cidrs in the list.
+
+    :param cidr: Cidr to check overlap for.
```

Comment:
This was passing in CIDR string values for this param and the list of CIDRs.  I think this makes sense since we have cidr strings passed in and the pool.cidr value is a string cidr.